File: API/Models/Product.cs
namespace API.Models
{
    public class Product:BaseModel
    {
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public ICollection<ProductImage> ProductImages { get; set; }
        public Category Category { get; set; }
        public int CategoryId { get; set; }
        public Brand Brand { get; set; }
        public int BrandId { get; set; }
    }
}

File: Frontend/client/src/app/shared/interfaces/Product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  quantity: number;
  imageUrl: string;
  productImages: string[];
  category: string;
  brand: string;

 // Optional fields for future features
  rating?: number;
  reviewCount?: number;
  isOnSale?: boolean;
  discountPercent?: number;
  inStock?: boolean;
  isInWishlist?: boolean;
  originalPrice?: number;
  freeShipping?: boolean;
}
